tests/test_inclined_exponential.py

```diff
@@ -0,0 +1,258 @@
+    
+
+@timer
+def test_sanity():
+    """ Performs various sanity checks on a set of InclinedExponential profiles. """
+    
+    for flux, inc_angle, scale_radius, scale_height, pos_angle in zip(fluxes,
+                                                                      image_inc_angles,
+                                                                      image_scale_radii,
+                                                                      image_scale_heights,
+                                                                      image_pos_angles):
+        
+        # Get float values for the details
+        flux = float(flux)
+        inc_angle=float(inc_angle)
+        scale_radius=float(scale_radius)/oversampling
+        scale_height=float(scale_height)/oversampling
+        pos_angle=float(pos_angle)
+        
+        # Now make a test image
+        test_profile = galsim.InclinedExponential(inc_angle*galsim.radians,scale_radius,scale_height,flux,
+                                                  gsparams=galsim.GSParams(maximum_fft_size=5000))
+        
+        # Rotate it by the position angle
+        test_profile = test_profile.rotate(pos_angle*galsim.radians)
+        
+        # Check that the k value for (0,0) is the flux
+        np.testing.assert_almost_equal(test_profile.kValue(kx=0.,ky=0.),flux)
+        
+        # Check that the drawn flux for a large image is indeed the flux
+        test_image = galsim.Image(5*image_nx,5*image_ny,scale=1.0)
+        test_profile.drawImage(test_image)
+        test_flux = test_image.array.sum()
+        np.testing.assert_almost_equal(test_flux,flux,decimal=3)
+        
+        # Check that the centroid is (0,0)
+        centroid = test_profile.centroid()
+        np.testing.assert_equal(centroid.x, 0.)
+        np.testing.assert_equal(centroid.y, 0.)
+        
+@timer
+def test_k_limits():
+    """ Check that the maxk and stepk give reasonable results for a few different profiles. """
+    
+    for inc_angle, scale_radius, scale_height in zip(image_inc_angles,image_scale_radii,
+                                                     image_scale_heights):
+        # Get float values for the details
+        inc_angle=float(inc_angle)
+        scale_radius=float(scale_radius)/oversampling
+        scale_height=float(scale_height)/oversampling
+        
+        gsparams = galsim.GSParams(maximum_fft_size=5000)
+    
+        # Now make a test image
+        test_profile = galsim.InclinedExponential(inc_angle*galsim.radians,scale_radius,scale_height,
+                                                  gsparams=gsparams)
+        
+        # Check that the k value at maxK() is below maxk_threshold in both the x and y dimensions
+        kx = test_profile.maxK()
+        ky = test_profile.maxK()
+        
+        kx_value=test_profile.kValue(kx=kx,ky=0.)
+        np.testing.assert_(np.abs(kx_value)<gsparams.maxk_threshold)
```

Comment:
I added print statements here to see how kValue(maxk) actually compares with maxk_threshold, and you are using much too large a value of maxk:
```
kvalue(maxk) =  (1.38421755486e-06+0j)
maxk_threshold =  0.001
```
The problem is you are still basing `maxk` off of where kval falls below `kvalue_accuracy`, rather than `maxk_threshold`.  

This is why you found it necessary to impose such a large `maximum_fft_size` in these tests (20,000 in `test_edge_on`!)  If you change that calculation to be based on `maxk_threshold`, you won't need such large ffts.